src/main/java/picard/fingerprint/FingerprintUtils.java

```diff
@@ -138,18 +137,31 @@ public static VariantContextSet createVCSetFromFingerprint(final Fingerprint fin
 
         // convert all the haplotypes to variant contexts and add them to the set.
         fingerPrint.values().stream()
-                .map(hp -> getVariantContext(reference, sample, hp))
+                .flatMap(hp -> getVariantContext(reference, sample, hp, representativeOnly))
                 .forEach(variantContexts::add);
 
         return variantContexts;
     }
 
     @VisibleForTesting
-    static VariantContext getVariantContext(final ReferenceSequenceFile reference,
+    static Stream<VariantContext> getVariantContext(final ReferenceSequenceFile reference,
                                                     final String sample,
-                                                    final HaplotypeProbabilities haplotypeProbabilities) {
-        final Snp snp = haplotypeProbabilities.getRepresentativeSnp();
-        final byte refAllele = StringUtil.toUpperCase(reference.getSubsequenceAt(
+                                                    final HaplotypeProbabilities haplotypeProbabilities,
+                                                    final boolean representative_only) {
+        if (representative_only) {
```

Comment:
I like flatmaps, it's a bit awkward how you have to repeat the same if/else logic inside the flatmap repeatedly though. You might pull out the HP -> stream<> logic into it's own method that you flatmap over, and then map the resulting stream to get the VariantContext or name as appropriate.  It looks like you probably didn't do that because of the way that getVariantContext relies on the outer HP so it can't be decomposed as nicely without nesting logic within the flatmap.  Do what you like with it.

Reply:
yeah...seems like over-complication...